core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1072,6 +1072,8 @@ private void waitForTableStateTransition(String tableId, TableState expectedStat
         if (Tables.getTableState(context.getInstance(), tableId) != expectedState) {
           if (!Tables.exists(context.getInstance(), tableId))
             throw new TableDeletedException(tableId);
+          if (Tables.getTableState(context.getInstance(), tableId) == TableState.DELETING)
```

Comment:
After clearing the cache could store the return value of getTableState in a local var (could call it `currentState`).  Then could avoid two calls to getTableState after clearing the cache.

Reply:
I thought about that but then was thinking that the state could change in that time.

Reply:
To properly account for changes between the times the two calls are made a loop would be needed.  However the wider code is in a loop, so its not needed here.

Reply:
OK I gotcha.  I didn't fully grasp what this loop and method was doing in the first place.